vignettes/04-synthesis-data.Rmd

```diff
@@ -1 +1,210 @@
+library(purrr)
+library(broom)
+
+options(betydb_url = "https://terraref.ncsa.illinois.edu/bety/",
+        betydb_api_version = 'v1', 
+        betydb_key = '9999999999999999999999999999999999999999')
+```
+
+```{r get_trait_data, message = FALSE}
+trait <- betydb_query(table     = "search", 
+                      trait     = "canopy_height", 
+                      date      = "~2017",
+                      limit     =  "none")
+
+trait = trait %>% 
+  mutate(day = as.Date(raw_date))
+```
+
+```{r get_weather_data}
+weather <- fromJSON('https://terraref.ncsa.illinois.edu/clowder/api/geostreams/datapoints?stream_id=46431&since=2017-01-01&until=2017-12-31', flatten = FALSE)
+
+weather <- weather$properties %>% 
+  mutate(time = ymd_hms(weather$end_time))
+
+daily_values = weather %>% 
+  mutate(day = as.Date(time), 
+         air_temp_converted = air_temperature - 273.15) %>% 
+  group_by(day) %>% 
+  summarise(min_temp = min(air_temp_converted), 
+            max_temp = max(air_temp_converted), 
+            gdd = ifelse(sum(min_temp, max_temp) / 2 > 10, 
+                         (max_temp + min_temp) / 2 - 10, 0))
+
+daily_values <- daily_values %>% 
+  mutate(gdd_cum = cumsum(gdd))
+```
+
+```{r combine_trait_weather}
+trait_weather_df = full_join(trait, daily_values, by = "day")
+trait_weather_df = trait_weather_df %>% 
+  select(day, cultivar, mean, gdd_cum) %>% 
+  na.omit()
+```
+
+## Plot and model relationship between GDD and canopy cover for each cultivar
+
+We are interested in how growing degree days affects canopy cover. 
+To investigate this, we are going to model and plot their relationship. 
+We want to know the relationship for each cultivar, so we'll start of by determining the parameters of the model for one of the cultivars in our dataset. 
+We are using a logistic growth model here because it is appropriate for the shape of the GDD-cover relationship. 
+
+```{r model_get_parameters}
+single_cultivar <- trait_weather_df %>% 
+  filter(cultivar == "PI656026")
+
+cap     <- 150
+initial <- 25
+mean    <- single_cultivar$mean[67]
+gdd_cum <- single_cultivar$gdd_cum[67]
+rate    <- ((log((cap/mean) - 1)) - initial)/gdd_cum
+
+model_single_cultivar <- nls(mean ~ cap / (1 + exp(initial + rate * gdd_cum)), 
+                             start = list(cap = cap, initial = initial, rate = rate),
+                             data = single_cultivar, trace = TRUE)
+
+single_cultivar <- single_cultivar %>% 
+  mutate(mean_predict = coef(model_single_cultivar)[1] / (1 + exp(coef(model_single_cultivar)[2] + coef(model_single_cultivar)[3] * gdd_cum)))
+```
+
+We then use the parameters from a single cultivar to run a model for each of the rest of the cultivars. 
+These results are used to plot the model predictions, which are shown as an orange line. 
+We also calculated the inflection point from each cultivar's model, which will be used in the following section. 
+
+```{r model_all_cultivars}
+model_parameters <- trait_weather_df %>% 
```

Comment:
you are correct - now that I look at it the for loop seems more straightforward, especially to those unfamiliar w/ the map function. A few thoughts:
1. to clarify you could 
  - feel free to return to the for loop or
  - add a few inline comments to explain what is happening 
1. eventually (not now) it would be better just to fit all data w/ random effects ... this is where I often leave it to the user

Reply:
I chose to revert this (22db799fdc2ddaf44e574f450f350d0fad4e1728). I'll go back to the nest/map method if I can figure out a more elegant way, I'm sure there is one.